src/certificate/commands/AbandonedCertificateCommand.ts

```diff
@@ -0,0 +1,62 @@
+import moment from 'moment';
+import { Service } from 'typedi';
+import { ICertificatePayload, TestResultSchemaTestTypesAsObject } from '../../models';
+import { CERTIFICATE_DATA, VEHICLE_TYPES } from '../../models/Enums';
+import { BasePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class AbandonedCertificateCommand extends BasePayloadCommand {
+	private certificateIsAbandoned = (): boolean => this.state.type === CERTIFICATE_DATA.ABANDONED_DATA;
+
+	public async generate(): Promise<ICertificatePayload> {
+		const result = {} as ICertificatePayload;
+		if (!this.certificateIsAbandoned()) {
+			return {} as ICertificatePayload;
+		}
+
+		const { testResult } = this.state;
+
+		const payload = await this.getPayloadData(testResult);
+
+		result.ABANDONED_DATA = {
+			...payload,
+		};
+
+		return result as ICertificatePayload;
+	}
+
+	private async getPayloadData(testResult: TestResultSchemaTestTypesAsObject): Promise<any> {
```

Comment:
Not a blocker for this PR, but just one to bare in mind in the future - the use of `async` here is not required. 
There is no asynchronous action being performed by this method

Reply:
Good spot, updated and pushed.